cmd/snap/cmd_run.go

```diff
@@ -387,6 +387,10 @@ func createUserDataDirs(info *snap.Info) error {
 		return fmt.Errorf(i18n.G("cannot get the current user: %v"), err)
 	}
 
+	snapDir := filepath.Join(usr.HomeDir, dirs.UserHomeSnapDir)
+	if err := os.MkdirAll(snapDir, 0700); err != nil {
```

Comment:
don't we need to change setup_user_data instead?

Reply:
So that function in snap-confine creates $SNAP_USER_DATA, which is something like `home/ijohnson/snap/hello-world/4`, whereas this function is creating `/home/ijohnson/snap`. I think it probably doesn't hurt to have $SNAP_USER_DATA to be created with 0700 too, but I don't know that it's necessary since the parent directory is already 0700 now with this PR

Reply:
I see, sorry it took me a bit to recontextualize this.